components/automate-ui/src/app/services/chef-session/chef-session.service.ts

```diff
@@ -26,29 +30,34 @@ export interface ChefSessionUser {
 }
 
 const sessionKey = 'chef-automate-user';
-const HTTP_STATUS_OK = 200;
 const HTTP_STATUS_UNAUTHORIZED = 401;
-const XHR_DONE = 4;
 
 // TODO 2017/11/06 sr: ChefSessionService implementing the route guard is a bit
 // of a hack. Something created for the purpose of route-guarding alone would
 // be better, and a good refactoring opportunity.
 @Injectable()
 export class ChefSessionService implements CanActivate {
   private user: ChefSessionUser;
+  private httpHandler: HttpClient;
+  private isRefreshing: boolean;
+  private tokenProvider: ReplaySubject<string>;
 
   // Session state keys - We use session storage to save state here because
   // the application can be reinitialized multiple time during a single session.
   //// Flag to store whether or not the modal has been displayed this session.
   //// Automatically set when the modal is shown for the first time.
   MODAL_HAS_BEEN_SEEN_KEY = 'welcome-modal-seen';
 
-  constructor(private store: Store<NgrxStateAtom>) {
+  constructor(private store: Store<NgrxStateAtom>, handler: HttpBackend) {
     // In dev mode, set a generic session so we don't
     // have to round-trip to the oidc provider (dex).
+    this.tokenProvider = new ReplaySubject(1);
```

Comment:
This works, but since you are always setting to 1, it is slightly cleaner to use a BehaviorSubject. It does exactly what a ReplaySubject-set-to-1 does.

Reply:
I initially had BehaviorSubject, but it requires you to pass an initial value which I think i do not want

Reply:
Ah, ok.